Link error: unresolved external symbol ___atomic_compare_exchange_8, with Clang-cl in Win32.

```text
1>[1/2] CXX obj/test/test_atomic/test_atomic/test_atomic.obj
1>[2/2] LINK test_atomic.exe test_atomic.exe.pdb
1>FAILED: test_atomic.exe test_atomic.exe.pdb
1>ninja -t msvc -e environment_x86 -- link.exe /nologo "-libpath:..\..\..\..\..\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.16.27023\ATLMFC\lib\x86" "-libpath:..\..\..\..\..\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.16.27023\lib\x86" "-libpath:..\..\..\..\..\Program Files (x86)\Windows Kits\10\lib\10.0.17763.0\ucrt\x86" "-libpath:..\..\..\..\..\Program Files (x86)\Windows Kits\10\lib\10.0.17763.0\um\x86" /OUT:./test_atomic.exe /PDB:./test_atomic.exe.pdb @./test_atomic.exe.rsp
1>test_atomic.obj : error LNK2019: unresolved external symbol ___atomic_compare_exchange_8 referenced in function _main
1>./test_atomic.exe : fatal error LNK1120: 1 unresolved externals
```

The problem happens in windows 10, using `Clang-cl` in `Win32`:

test\test_atomic\test_atomic.cpp
```C++
#include <atomic>
#include <cinttypes>

int main(void) {
  std::atomic<std::int64_t> temp;
  temp.load();
}
```

test\test_atomic\build.gn
```gn
executable("test_atomic") {
  sources = [
    "test_atomic.cpp",
  ]
}
```
gn && ninja command line
```cmd
gn gen out/llvm_win32_release --ide=vs2017 --winsdk=10.0.17763.0 --args="clang_base_path=\"C:/Program Files/LLVM\" is_clang=true target_cpu=\"x86\" is_debug=false"

ninja -C out/llvm_win32_release test_atomic

```

PS: The problem just happens in `LLVM+Win32`. There are no problem with `LLVM+Win64`, `Win32` and `Win64`.

--- a/config/win/BUILD.gn
+++ b/config/win/BUILD.gn
@@ -62,7 +62,9 @@ config("compiler") {
   if (current_cpu == "x86") {
     # TODO: Move this somewhere else (compiler_codegen?)
     # Preserve previous /arch behaviour (since 12.0 it defaults to /arch:SSE2)
-    cflags += [ "/arch:IA32" ]
+    # The options would cause link error, in `LLVM+Win32`: 
+    # unresolved external symbol ___atomic_compare_exchange_8
+    # cflags += [ "/arch:IA32" ]
   }
 
   # Force C/C++ mode for the given GN detected file type. This is necessary

